airflow/utils/log/logging_mixin.py

```diff
@@ -107,6 +110,34 @@ def isatty(self):
         return False
 
 
+class RedirectStdHandler(StreamHandler):
+    """
+    Handler that allows stdout/stderr to be redirected after initalization
+    """
+    def __init__(self, stream):
+        if not isinstance(stream, six.string_types):
+            raise Exception("Cannot use file like objects. Use 'stdout' or 'stderr'"
+                            " as a str and without 'ext://'.")
+
+        self._use_stderr = True
+        if 'stdout' in stream:
+            self._use_stderr = False
+            stream = sys.stdout
+        else:
+            stream = sys.stderr
+
+        super(RedirectStdHandler, self).__init__(stream)
+
+    def emit(self, record):
+        # This makes sure we use the current sys.stdxxx in case it is being redirected
+        if self._use_stderr:
+            self.stream = sys.stderr
+        else:
+            self.stream = sys.stdout
```

Comment:
Rather than setting self.stream each time maybe do something like this:

```python
    @property
    def stream(self):
        if self._use_stderr:
            return sys.stderr
        return sys.stdout
```

(Inspiration from https://github.com/python/cpython/blob/3.6/Lib/logging/__init__.py#L1077-L1095)

Reply:
Oh I like that :)